### PR TITLE
Godeps: Bump rkt api, go grpc, go protobuf.

### PR DESCRIPTION
**Pull request by yifan-gu**
As a result of grpc bump, golang.org/x/net/http2 is added.

How I run to update the godep:
- godep restore
- checkout etcd, rkt, go grpc, go protobuf, go go protobuf to latest
- rm -fr Godeps
- godep save ./...
- hack/update-godep-licenses.sh

### CODE REVIEW

**Comment by mikedanese** [comment]
@k8s-bot ok to test

**Comment by yifan-gu** [comment]
From build log:
```
Godeps/_workspace/src/github.com/coreos/etcd/etcdserver/etcdserverpb/rpc.pb.go:534: cannot use _Etcd_Range_Handler (type func(interface {}, context.Context, grpc.Codec, []byte) (interface {}, error)) as type grpc.methodHandler in field value
Godeps/_workspace/src/github.com/coreos/etcd/etcdserver/etcdserverpb/rpc.pb.go:538: cannot use _Etcd_Put_Handler (type func(interface {}, context.Context, grpc.Codec, []byte) (interface {}, error)) as type grpc.methodHandler in field value
Godeps/_workspace/src/github.com/coreos/etcd/etcdserver/etcdserverpb/rpc.pb.go:542: cannot use _Etcd_DeleteRange_Handler (type func(interface {}, context.Context, grpc.Codec, []byte) (interface {}, error)) as type grpc.methodHandler in field value
Godeps/_workspace/src/github.com/coreos/etcd/etcdserver/etcdserverpb/rpc.pb.go:546: cannot use _Etcd_Txn_Handler (type func(interface {}, context.Context, grpc.Codec, []byte) (interface {}, error)) as type grpc.methodHandler in field value
Godeps/_workspace/src/github.com/coreos/etcd/etcdserver/etcdserverpb/rpc.pb.go:550: cannot use _Etcd_Compact_Handler (type func(interface {}, context.Context, grpc.Codec, []byte) (interface {}, error)) as type grpc.methodHandler in field value
```

I suspect etcd is using an older version go grpc. @mikedanese

**Comment by yifan-gu** [comment]
cc @jonboulle @xiang90 @gyuho

**Comment by gyuho** [comment]
@yifan-gu We can bump up the gRPC. Which latest version gRPC is compatible with kubernetes?

**Comment by yifan-gu** [comment]
@gyuho In rkt we updated the grpc version recently, and use that to generate the code.
 The commit hash is:
https://github.com/coreos/rkt/blob/master/Godeps/Godeps.json#L360

**Comment by gyuho** [comment]
@yifan-gu Ok, I will try to match that in etcd.

**Comment by yifan-gu** [comment]
@gyuho Thank you :)

**Comment by yifan-gu** [comment]
@k8s-bot test this please

**Comment by jonboulle** [comment]
@yifan-gu do we also need to bump etcd in the same commit now ..?

**Comment by yifan-gu** [comment]
> @yifan-gu do we also need to bump etcd in the same commit now ..?

Yes we need to include the recent etcd commits which bump the grpc.

Tests are still failing, ehh, investigating.

**Comment by yifan-gu** [comment]
OOB discussed with @xiang90 . Will wait for etcd to back port the grpc commits to stable branch, and then I will bump etcd to the stable branch.

**Comment by jonboulle** [comment]
This should be good to go with etcd 2.2.5: https://github.com/coreos/etcd/releases/tag/v2.2.5

**Comment by yifan-gu** [comment]
> This should be good to go with etcd 2.2.5: https://github.com/coreos/etcd/releases/tag/v2.2.5

ACK, will update today. Thanks @jonboulle !

**Comment by yifan-gu** [comment]
Updated with etcd 2.2.5

**Comment by yifan-gu** [comment]
@k8s-bot test this please

**Comment by yifan-gu** [comment]
Ready for another review.

File: Godeps/Godeps.json [hunk 1]
```diff
@@ -1041,7 +1041,7 @@
 		},
 		{
 			"ImportPath": "k8s.io/heapster/api/v1/types",
-			"Comment": "v0.19.1",
+			"Comment": "v0.19.1-44-g0991ac5",
```
Comment (mikedanese) [inline Godeps/Godeps.json]: cc @vishh

**Comment by mikedanese** [comment]
Closing for #20831

**Pull request closed without merging**